src/SignalR/clients/csharp/Client/test/UnitTests/LongPollingTransportTests.cs

```diff
@@ -692,4 +693,45 @@ public async Task SendsDeleteRequestWhenTransportCompleted()
             Assert.Equal(TestUri, deleteRequest.RequestUri);
         }
     }
+
+    [Fact]
+    public async Task StartAsyncSetsAcceptHeaderCorrectly()
+    {
+        var mockHttpHandler = new Mock<HttpMessageHandler>();
+        var responseTaskCompletionSource = new TaskCompletionSource<HttpResponseMessage>();
+
+        mockHttpHandler.Protected()
+            .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+            .Returns((HttpRequestMessage request, CancellationToken cancellationToken) =>
+            {
+                if (request.Headers.Accept?.Contains(new MediaTypeWithQualityHeaderValue("*/*")) == true)
+                {
+                    responseTaskCompletionSource.SetResult(ResponseUtils.CreateResponse(HttpStatusCode.OK));
```

Comment:
Isn't this going to be hit twice? Which would throw internally I think?

Reply:
I believe you are correct; the `TaskCompletionSouce` can only be completed once. I have refactored the test with an improved methodology.